plugins/modules/foreman_scc_account.py

```diff
@@ -0,0 +1,177 @@
+    choices: ["present", "absent", "synced"]
+    type: str
+extends_documentation_fragment:
+  - foreman
+  - foreman.organization
+'''
+
+EXAMPLES = '''
+- name: "Create an scc_account"
+  foreman_scc_account:
+    username: "admin"
+    password: "changeme"
+    server_url: "https://foreman.example.com"
+    name: "Test"
+    login: "abcde"
+    scc_account_password: "12345"
+    base_url: "https://scc.suse.com"
+    state: present
+
+- name: "Update a scc_account"
+  foreman_scc_account:
+    username: "admin"
+    password: "changeme"
+    server_url: "https://foreman.example.com"
+    name: "Test1"
+    state: present
+
+- name: "Delete a scc_account"
+  foreman_scc_account:
+    username: "admin"
+    password: "changeme"
+    server_url: "https://foreman.example.com"
+    name: "Test"
+    state: absent
+'''
+
+RETURN = ''' # '''
+
+
+from ansible.module_utils.foreman_helper import KatelloEntityAnsibleModule
+
+
+def main():
+    module = KatelloEntityAnsibleModule(
+        entity_spec=dict(
+            name=dict(required=True),
+            updated_name=dict(),
+            login=dict(),
+            scc_account_password=dict(no_log=True, flat_name='password'),
+            base_url=dict(),
+            sync_date=dict(),
+            interval=dict(choices=['never', 'daily', 'weekly', 'monthly']),
+            state=dict(default='present', choices=['present', 'absent', 'synced']),
+        ),
+        argument_spec=dict(
+            test_connection=dict(type='bool', default=False)),
+    )
+
+    entity_dict = module.clean_params()
+
+    module.connect()
+
+    entity_dict, scope = module.handle_organization_param(entity_dict)
+
+    entity = module.find_resource_by_name('scc_accounts', name=entity_dict['name'], params=scope, failsafe=True)
+
+    if not module.desired_absent:
+        if not entity:
+            if 'login' not in entity_dict:
+                module.fail_json(msg="scc account login not provided")
+            if 'scc_account_password' not in entity_dict:
+                module.fail_json(msg="Scc account password not provided")
+
+        if entity_dict['test_connection']:
+            scc_account_credentials = {}
+            if entity:
+                scc_account_credentials['id'] = entity['id']
+            if 'login' in entity_dict:
+                scc_account_credentials['login'] = entity_dict['login']
+            if 'scc_account_password' in entity_dict:
+                scc_account_credentials['password'] = entity_dict['scc_account_password']
+            if 'base_url' in entity_dict:
+                scc_account_credentials['base_url'] = entity_dict['base_url']
+            module.resource_action('scc_accounts', 'test_connection', scc_account_credentials, ignore_check_mode=True)
+
+    if 'updated_name' in entity_dict:
+        entity_dict['name'] = entity_dict['updated_name']
+
+    if module.state == 'synced':
+        module.resource_action('scc_accounts', 'sync', {'id': entity['id']})
```

Comment:
this will fail if `entity` is `None`, which it can be as you searched with `failsafe=True` above.
```suggestion
        if entity:
            module.resource_action('scc_accounts', 'sync', {'id': entity['id']})
        else:
            module.fail_json(msg="Could not find SCC account {0} to sync".format(entity_dict['name']))
```

Reply:
Sorry for being late to the party. But i just realized, we can use `failsafe= (module.state != 'synched')`.
I'm not going to block on this one.

Reply:
Agreed, feel free to add that. ACK either way holds.